### ISSUE TITLE
Improve color controls for flutter.

### ISSUE THREAD

**Issue by MichealReed** [flutter/flutter#37865]
It would be great to have improved color functions similar to as outlined here:

https://thoughtbot.com/blog/controlling-color-with-sass-color-functions

**Comment by shyndman** [comment]
Check out [`HSLColor`](https://api.flutter.dev/flutter/painting/HSLColor-class.html) and [`HSVColor`](https://api.flutter.dev/flutter/painting/HSVColor-class.html). They can do all of those things.

**Comment by goderbauer** [comment]
@MichealReed Did that answer your question?

**Comment by MichealReed** [comment]
No, I am familiar with the functions available and while yes, you can accomplish what I am asking,  the approach sass takes is much more user friendly. Only a suggestion.
